middleware/pprof/pprof.go

```diff
@@ -38,43 +41,53 @@ func New(config ...Config) fiber.Handler {
 
 		path := c.Path()
 		// We are only interested in /debug/pprof routes
-		if len(path) < 12 || !strings.HasPrefix(path, cfg.Prefix+"/debug/pprof") {
+		path, found := cutPrefix(path, prefix)
+		if !found {
 			return c.Next()
 		}
-		// Switch to original path without stripped slashes
+		// Switch on trimmed path against constant strings
 		switch path {
-		case cfg.Prefix + "/debug/pprof/":
+		case "/":
 			pprofIndex(c.Context())
-		case cfg.Prefix + "/debug/pprof/cmdline":
+		case "/cmdline":
 			pprofCmdline(c.Context())
-		case cfg.Prefix + "/debug/pprof/profile":
+		case "/profile":
 			pprofProfile(c.Context())
-		case cfg.Prefix + "/debug/pprof/symbol":
+		case "/symbol":
 			pprofSymbol(c.Context())
-		case cfg.Prefix + "/debug/pprof/trace":
+		case "/trace":
 			pprofTrace(c.Context())
-		case cfg.Prefix + "/debug/pprof/allocs":
+		case "/allocs":
 			pprofAllocs(c.Context())
-		case cfg.Prefix + "/debug/pprof/block":
+		case "/block":
 			pprofBlock(c.Context())
-		case cfg.Prefix + "/debug/pprof/goroutine":
+		case "/goroutine":
 			pprofGoroutine(c.Context())
-		case cfg.Prefix + "/debug/pprof/heap":
+		case "/heap":
 			pprofHeap(c.Context())
-		case cfg.Prefix + "/debug/pprof/mutex":
+		case "/mutex":
 			pprofMutex(c.Context())
-		case cfg.Prefix + "/debug/pprof/threadcreate":
+		case "/threadcreate":
 			pprofThreadcreate(c.Context())
 		default:
 			// pprof index only works with trailing slash
 			if strings.HasSuffix(path, "/") {
 				path = strings.TrimRight(path, "/")
 			} else {
-				path = cfg.Prefix + "/debug/pprof/"
+				path = prefix + "/"
 			}
 
 			return c.Redirect(path, fiber.StatusFound)
 		}
 		return nil
 	}
 }
+
+// cutPrefix is a copy of [strings.CutPrefix] added in Go 1.20.
+// Remove this function when we drop support for Go 1.19.
+func cutPrefix(s, prefix string) (after string, found bool) {
```

Comment:
The `nonamedreturns` linter does not like its signature. Should I drop the names to make the linter happy, or add `//nolint:nonamedreturns` so it stays the same with its equivalent in std?

Reply:
@leonklingele what do you think? 
golang is also using named returns https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/strings/strings.go;l=1289

Reply:
Went ahead with `//nolint:nonamedreturns`.

Reply:
Updated the comment to make the linter happy.